src/test/java/org/sonarsource/sonarlint/ls/mediumtests/NotebookMediumTests.java

```diff
@@ -61,4 +67,24 @@ void analyzeNotebookOnOpen() throws Exception {
     assertThat(client.getDiagnostics(uri)).isEmpty();
   }
 
+  @Test
+  void shouldLogIllegalStateErrorWhenDidChangeReceivedAndNotebookIsNotOpen() {
+    var notebookUri = URI.create("file:///some/notebook.ipynb");
+
+    var cell1 = new TextDocumentItem();
+    cell1.setUri(notebookUri + "#cell1");
+    cell1.setText("cell1 line1\ncell1 line2\n");
+
+    var cell2 = new TextDocumentItem();
+    cell2.setUri(notebookUri + "#cell2");
+    cell2.setText("cell2 line1\ncell2 line2\n");
+
+    setShowVerboseLogs(client.globalSettings, true);
+    var logger = mock(LanguageClientLogger.class);
+    doCallRealMethod().when(logger).warn(any());
+
+    didChange(notebookUri.toString(), "newContent");
```

Comment:
@sophio-japharidze-sonarsource if I'm not mistaken, this call uses the LSP's `TextDocumentService`, not the `NotebookDocumentService` - which could explain why the "notebook" path is not marked as covered.